Client/multiplayer_sa/CMultiplayerSA.cpp

```diff
@@ -324,6 +324,32 @@ BYTE  ucSkyGradientTopB = 0;
 BYTE  ucSkyGradientBottomR = 0;
 BYTE  ucSkyGradientBottomG = 0;
 BYTE  ucSkyGradientBottomB = 0;
+
+bool  bUsingCustomAmbientColor = false;
+float fAmbientColorR = 0.0F;
+float fAmbientColorG = 0.0F;
+float fAmbientColorB = 0.0F;
+
+bool  bUsingCustomAmbientObjectColor = false;
+float fAmbientObjectColorR = 0.0F;
+float fAmbientObjectColorG = 0.0F;
+float fAmbientObjectColorB = 0.0F;
+
+bool  bUsingCustomDirectionalColor = false;
+float fDirectionalColorR = 0.0F;
+float fDirectionalColorG = 0.0F;
+float fDirectionalColorB = 0.0F;
+
+bool  bUsingCustomLowCloudsColor = false;
+int16 iLowCloudsColorR = 0;
+int16 iLowCloudsColorG = 0;
+int16 iLowCloudsColorB = 0;
+
+bool  bUsingCustomBottomCloudsColor = false;
+int16 iBottomCloudsColorR = 0;
+int16 iBottomCloudsColorG = 0;
+int16 iBottomCloudsColorB = 0;
```

Comment:
`std::int16_t` instead of `int16`

Reply:
Technically `std::int16_t` and `int16` are the same (short is signed by default) 🤔
`int16` is more clean and consistent with reversed code

Reply:
> Technically `std::int16_t` and `int16` are the same (short is signed by default) 🤔 `int16` is more clean and consistent with reversed code

No. They are **not** the same. `std::int16_t` is crossplatform while `int16` is not. Please use `std` variant instead

Reply:
I'd use it for shared or server-side code. But this code is strictly client-side and not cross-platform. It uses the same definition on Windows.

To be honest with you, I don't agree with almost all code suggestions from you. It's not necessary at all and breaks consistency in modified files. I decided to make some changes only because I'm polite and this useful PR was ignored for a year.
For example I don't want to use #define for memory addresses in this particular PR. I strongly oppose not consistent code in files because it makes it harder to work with in the long run. If this is actually required to merge this PR then I'd like to get a comment from Member / someone who is interested in merging this PR.

Reply:
So, you'd rather stick to unknown addresses without documenting them...?
Its alright. Your PR, your call, just trying to make the source code more reliable and less messy

Reply:
Maybe let's make a small compromise and just add comments above the addresses for now?

Reply:
Its alright. If it would bother other people then a refactoring PR will be posted.

Reply:
It's documented in reversed code. And this is actually the only place where it's useful.

There is a reason why other similar functions almost don't use definitions for addresses. For example you don't #define addresses when you NOP the same variable in different places because it makes code messy and hard to work with.
It makes sense to #define the address for overwritten variable. But you can also just put a new line there and keep this overwritten variable separated. And this is actually how it's used in all similar functions right now because it's easy to read and compare with reversed code.

Reply:
By reversed code you mean...?

Reply:
> By reversed code you mean...?

I'm obviously talking about private `.idb` files of `gta_sa.exe` used for creating features like this.

Reply:
> > By reversed code you mean...?
> 
> I'm obviously talking about private `.idb` files of `gta_sa.exe` used for creating features like this.

Thought you were talking about gta-reversed for a moment there. Its okay 👍